### ISSUE TITLE
Multiple Parsing errors in Zotero Reader.

### ISSUE THREAD

**Issue by nischalsamji** [diging/tethne#85]
The Zotero reader has multiple parsing errors while encoding/ decoding unicode data. 

Error occurred while trying to import a zotero corpus using tethne/readers/zotero.read()
Test data provided on request

error log:

``` python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "zotero.py", line 180, in read
    papers = ZoteroParser(path).parse()
  File "/home/nischal/diging/tethne/tethne/readers/base.py", line 278, in parse
    self.handle(tag, o)    
  File "/home/nischal/diging/tethne/tethne/readers/base.py", line 287, in handle
    data = handler(data)
  File "zotero.py", line 96, in handle_authors_full
    in self.graph.triples((value, None, None))]        
  File "zotero.py", line 106, in handle_author
    forename = str([e[2] for e in forename_iter][0]).upper().replace('.', '')
  File "/home/nischal/.virtualenvs/myenv/local/lib/python2.7/site-packages/rdflib/term.py", line 1262, in __str__
    return self.encode()
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 1: ordinal not in range(128)
```

**Comment by erickpeirson** [comment]
@nischalsamji, encoding is the bane of my existence. We should develop a solution that fixes this problem in a systematic way.

**Comment by nischalsamji** [comment]
@erickpeirson Every developer's nightmare !!! Exploring different solutions, Will discuss with others and hopefully fix this thing !!!

**Comment by erickpeirson** [comment]
@nischalsamji Excellent. Look for the point at which the file/buffer is opened. Depending on how the file pointer is getting created, there should be a simple way to set the encoding. That's probably the most straight-forward approach, unless there is something strange going on.

**Comment by nischalsamji** [comment]
@erickpeirson Think the issue is from rdflib used to parse .rdf files.

**Comment by erickpeirson** [comment]
Hmmm, can you say more about what you think is going wrong?

---

Erick Peirson
Postdoctoral Scholar
ASU-SFI Center for Biosocial Complexity
Arizona State University

**Comment by nischalsamji** [comment]
@erickpeirson Need help on this issue. As you can see, the last line of the stacktrace 

``` python

File "/home/nischal/.virtualenvs/myenv/local/lib/python2.7/site-packages/rdflib/term.py", line 1262, in __str__
    return self.encode()
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 1: ordinal not in range(128)
```

indicates that the error is in rdflib. Am I going in the right direction.

**Comment by erickpeirson** [comment]
@nischalsamji Well, the error originates in rdflib, but that may not be the ultimate source of the problem. Take a close look at the error:

`...line 1282, in __str__`

That means that the problem arises when we attempt to coerce something (whatever class owns that chunk of code) to a str, e.g. by doing something like `str(someobj)`.

So, take a look at the line just above it; do you see where that might be happening?

**Comment by nischalsamji** [comment]
Yes !!! I fixed the issue. Writing tests now.

**Comment by erickpeirson** [comment]
Excellent; good work!

**Comment by nischalsamji** [comment]
@erickpeirson I need to run the existing tests for a new dataset i.e a new .rdf file. I am planning to do the tests in this way...

``` python
for each_rdffile in rdffiles:
     do_tests()
```

Is this is a good approach? Is there a better way to do it?

**Comment by erickpeirson** [comment]
This is fixed in pr #96
